OP2Helper.cpp

```diff
@@ -54,15 +54,21 @@ void CreateTubeOrWallLine(int x1, int y1, int x2, int y2, map_id type)
 	horz = y1;
 
 	// Make sure (x1 <= x2) and (y1 <= y2)
-	if (x1 > x2){ x1 ^= x2; x2 ^= x1; x1 ^= x2;	}
-	if (y1 > y2){ y1 ^= y2; y2 ^= y1; y1 ^= y2;	}
+	if (x1 > x2) { 
+		x1 ^= x2; x2 ^= x1; x1 ^= x2;
+	}
+	if (y1 > y2) { 
+		y1 ^= y2; y2 ^= y1; y1 ^= y2;
+	}

```

Comment:
Ahh yes, I was looking at this a few days ago and thought this code really needed to go. It's using the "clever" XOR swap algorithm, which is really not very readable, and not a good idea to use inline in code like this. It should be replaced by something like `std::swap(x1, x2)` and `std::swap(y1, y2)`. See [`std::swap`](https://en.cppreference.com/w/cpp/algorithm/swap). It seems there is a choice of header includes for `swap`.

Reply:
But how am I supposed to practice bit twiddling now???